README.md

```diff
@@ -124,6 +124,13 @@ spec:
 kubectl create -f aadpodidentity.yaml
 ```
 
+#### Understanding Namespaced identities
+The system will match `pod` to `identity` across namespaces by default. This behavior can be modified to match to pods within the namespace that holds `AzureIdentity` by
+
+1. On Azure Identity bases, by addinging `aadpodidentity.k8s.io/Behavior: namespaced` annotation (You have to add the annotation on each `AzureIdentity` you want to apply this behavior on).
+2. Default namespaced behavior on all identities by adding `--forceNamespaced=true` argument on the command line  or declare `FORCENAMESPACED=true` environment variable (for both `nmi` and `mic`).
```

Comment:
We should document performing 2 can override configurations set in 1 given this `if s.IsNamespaced || aadpodid.IsNamespacedIdenity(&val)` in mic.go

Reply:
I am not sure i understand this one, can you please elaborate?

Reply:
If `--forceNamespaced=true` is set as described in 2, then the namespaced behavior is set on all identities, regardless what annotation is used on the individual AzureIdentity. Hence setting 2 can override configurations set in 1. This behavior should be called out.

Reply:
@khenidak @ritazh  I understand the motivation for avoiding breaking changes, however, from a security standpoint, it would be better to have `forceNamespaced=false` by default. That would this application a bit more "secure by default", which would be easier for people to use it without needing to know much about the project. 

If the priority of not breaking compatibility is higher, we probably should emphasise in the documentation the potential lateral movement within Azure Control Plane, specially for cases of clusters hosting multiple applications/used by multiple teams.

Reply:
A hard break may be desirable but is dangerous from an end-user perspective. We should probably follow a formal deprecation policy and communicate after X time the default will be flipped to the more secure model.

Reply:
> from a security standpoint, it would be better to have forceNamespaced=false by default. That would this application a bit more "secure by default",

@pjbgf Perhaps I'm misunderstanding your point here but wouldnt `forceNamespaced=false` make it less secure since we are assigning the identity to any binding that match regardless of the namespace?
https://github.com/Azure/aad-pod-identity/blob/6ac452027c9246ec057a7b66c551c1e9c6e8c2ab/pkg/mic/mic.go#L185-L192

Reply:
`forceNamespaced=false` is the default. it is an opt in feature to maintain back combat. Please keep in mind that namespaced identities/binding/assignment is not a security feature per se, it is more of an organizational feature.

I want us to maintain back combat (unless we `absolutely` have to). I hate to ask users to modify anything (which may include an app downtime) they have already working.

Reply:
I'm a likely consumer of this capability. I don't have any existing deployments (so I don't personally care about backward compatibility). I don't perceive a problem with requiring me to specify forceNamespaced=true to get the behavior I want.

I perceive this to be a security feature; it makes me uneasy if you think it's just an organizational feature. I'm standing up AKS with AD integration (so I know who the users are), and using Kubernetes RBAC to restrict users to only work in authorized namespaces and with authorized object types. I perceive this integration to be a useful way for each namespace to be associated with an azure service principal so that user (and only that user) can securely connection to Azure resources outside the cluster that have potentially sensitive data.

Will using Azure pod identity and keyvault-flexvol introduce security concerns?

Reply:
@ritazh You are right, I meant to say `forceNamespaced=true`. I just edited that to make sense. 
I also highlighted [here](https://medium.com/@pjbgf/using-aad-pod-identity-in-your-azure-kubernetes-clusters-what-to-watch-out-for-73d5d73960f) three scenarios (1, 2 and 3) in which having `forceNamespaced=false` makes it easier for non-admins to do lateral movements between the cluster control plane and the azure control plane.